docs/concepts.md

```diff
@@ -0,0 +1,84 @@
+# Concepts
+
+Feldera Platform processes queries and produces output continuously.  When input
```

Comment:
Not sure if we should continue to stick to "Feldera Platform". Maybe "Feldera Engine" or "Feldera" in short? cc @ryzhyk